app/waf/src/models/as3.model.ts

```diff
@@ -1,387 +1,47 @@
-      }
-    }
-  }
-  toJSON(): Object {
-    let obj: AS3JSONObject = {
-      type: this.type,
-    };
-    if (this.type === 'waf') {
-      if (!isNullOrUndefined(this.wafpolicy)) {
-        obj.policy = this.wafpolicy;
-      }
-    }
-    return obj;
-  }
-}
-export class AS3Rule extends AS3Object {
-  name: string;
-  conditions: AS3Condition[];
-  actions: AS3Action[];
-
-  constructor(rule: Rule, params: {[key: string]: Object}) {
-    super();
-
-    this.conditions = [];
-    this.actions = [];
-
-    let conditions = <Condition[]>rule.conditions;
-    let actions = <Action[]>rule.actions;
-
-    if (rule.name) {
-      this.name = rule.name;
-    }
-    for (let condition of conditions) {
-      this.conditions.push(new AS3Condition(condition));
-    }
-
-    for (let action of actions) {
-      this.actions.push(new AS3Action(action, params));
-    }
-  }
-}
-export class AS3EndpointPolicy extends AS3Object {
-  name?: string;
-  rules: AS3Rule[];
-  constructor(params: {[key: string]: Object}) {
-    super();
-    let endpointpolicy = <Endpointpolicy>params['endpointpolicy'];
-    let rules = <Rule[]>params.rules;
-    this.class = 'Endpoint_Policy';
-    this.name = endpointpolicy.name;
-
-    this.rules = [];
-
-    for (let rule of rules) {
-      this.rules.push(new AS3Rule(rule, params));
-    }
-  }
-  toJSON(): Object {
-    let obj: AS3JSONObject = {
-      class: this.class,
-      rules: this.rules,
+    this.declaration = {
+      class: 'ADC',
+      schemaVersion: '3.0.0',
+      id: adc.id,
     };
-    return obj;
-  }
-}
-export class AS3Pool extends AS3Object {
-  name: string;
-  monitors: string[];
-  members: AS3PoolMember[];
-  constructor(params: {[key: string]: Object}) {
-    super();
-
-    this.class = 'Pool';
-    this.monitors = ['http'];
 
-    let pool = <Pool>params.pool;
-    if (pool.name) {
-      this.name = pool.name;
+    if (adc.name) {
+      this.declaration.label = adc.name;
     }
 
-    let members = <Member[]>params.members;
-
-    this.members = [];
-    for (let member of members) {
-      this.members.push(new AS3PoolMember(member));
+    if (adc.description) {
+      this.declaration.remark = adc.description;
     }
-  }
 
-  toJSON(): Object {
-    let obj: AS3JSONObject = {
-      class: this.class,
-      monitors: this.monitors,
-      members: this.members,
+    let tenant = 'F5_' + application.tenantId;
+    this.declaration[tenant] = {
+      class: 'Tenant',
     };
-    return obj;
-  }
-}
 
-export class AS3PoolMember extends Object {
-  servicePort: number;
-  serverAddresses: string[];
-
-  constructor(member: Member) {
-    super();
-
-    this.servicePort = member.port;
-    this.serverAddresses = [member.address];
-  }
-}
-
-export class AS3WAFPolicy extends AS3Object {
-  name: string;
-  url: string;
-
-  constructor(policy: Wafpolicy) {
-    super();
-
-    this.class = 'WAF_Policy';
-    if (policy.name) {
-      this.name = policy.name;
+    if (declaration) {
```

Comment:
if no declaration(declaration is undefined), throw exception directly. 
We don't allow deployment with no declaration go forwards.

Reply:
A request without declaration is the request to remove the tenant in BIG-IP. That is the behavior we need.